lib/improver/utilities/statistical_operations.py

```diff
@@ -0,0 +1,213 @@
+        type (e.g. height). This returns the percentile with which these
+        heights would be associated in the reference field. This effectively
+        uses the field of values as a 2-dimensional set of thresholds, and the
+        percentiles looked up correspond to the probabilities of these
+        thresholds being reached.
+
+        Snow-fall level::
+
+            Reference field: Percentiled snow fall level (m ASL)
+            Other field: Orography (m ASL)
+
+            300m ----------------- 30th Percentile snow fall level
+            200m ----_------------ 20th Percentile snow fall level
+            100m ---/-\----------- 10th Percentile snow fall level
+            000m --/---\----------  0th Percentile snow fall level
+            ______/     \_________ Orogaphy
+
+        The orography heights are compared against the heights that
+        correspond with percentile values to find the band in which they
+        fall; this diagram hides the 2-dimensional variability of the snow
+        fall level. The percentile values are then interpolated to the
+        height of the point being considered. This constructs a
+        2-dimensional field of probabilities that snow will be falling at
+        each point in the orography field.
+    """
+
+    def __init__(self, reference_cube, output_name, inverse_ordering=False):
+        """
+        Initialise class.
+
+        Args:
+            reference_cube (iris.cube.Cube):
+                The percentiled field from which probabilities will be obtained
+                using the input cube.
+            output_name (str):
+                The name of the cube being created,
+                e.g.'probability_of_snowfall'.
+        Keyword Args:
+            inverse_ordering (bool):
+                Set True if the percentiled data increases in the opposite
+                sense to the percentile coordinate.
+                e.g.  0th Percentile - Value = 10
+                     10th Percentile - Value = 5
+                     20th Percenitle - Value = 0
+        """
+        self.reference_cube = reference_cube
+        self.output_name = output_name
+        self.inverse_ordering = inverse_ordering
+
+    def create_probability_cube(self, cube):
+        """
+        Create a probability cube in which to store the calculated
+        probabilities.
+
+        Args:
+            cube (iris.cube.Cube):
+                The input cube to be used as a template.
+        Returns:
+            probability_cube (iris.cube.Cube):
+                The new cube with suitable metadata.
+        """
+        cube_format = next(cube.slices([cube.coord(axis='y'),
+                                        cube.coord(axis='x')]))
+        probabilities = cube_format.copy(data=np.full(cube_format.shape, 1.,
+                                                      dtype=float))
+        percentile_coordinate, = [coord.name() for coord in
```

Comment:
You are looking for percentile coordinate in several places within this code and I know it is being looked for elsewhere (e.g. blending). It would probably make sense to add a function to the utilities code to do this search and use this same bit of code everywhere. Not that this is a large bit of code but if we change the way we define a percentile coordinate we would then only have to change the search in the utility rather than everywhere within the code.